### ISSUE TITLE
hydra-unstable does not build on `master`

### ISSUE THREAD

**Issue by Taneb** [NixOS/nixpkgs#171406]
## Issue description

The `hydra-unstable` package fails to build on `nixos-unstable` and `master`. I'd like to be using Hydra from nixpkgs unstable on a personal server.

The error message looks like this:

```
g++ -DHAVE_CONFIG_H -I. -I../..    -std=c++17 -I/nix/store/73r0i5y56qvnhmzm5f9js4y59bi9apbl-boehm-gc-8.0.6-dev/include -I/nix/store/w9qnn4vw72fmiivd6p5hpiny504pz6yh-nix-2.8pre20220411_f7276bc-dev/include/nix -Wall -I ../libhydra -Wno-deprecated-declarations -g -O2 -std=c++17 -include nix/config.h -c -o hydra_evaluator-hydra-evaluator.o `test -f 'hydra-evaluator.cc' || echo './'`hydra-evaluator.cc
In file included from hydra-evaluator.cc:1:
../libhydra/db.hh: In member function 'std::string Connection::getFlags()':
../libhydra/db.hh:21:66: error: 'string' was not declared in this scope
   21 |             return concatStringsSep(" ", tokenizeString<Strings>(string(s, lower_prefix.size()), ";"));
      |                                                                  ^~~~~~
../libhydra/db.hh:21:66: note: suggested alternatives:
In file included from /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/string:39,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/internal/encodings.hxx:16,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array.hxx:18,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array:4,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/pqxx:2,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
/nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/bits/stringfwd.h:79:33: note:   'std::string'
   79 |   typedef basic_string<char>    string;
      |                                 ^~~~~~
In file included from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/internal/encodings.hxx:16,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array.hxx:18,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array:4,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/pqxx:2,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
/nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/string:67:11: note:   'std::pmr::string'
   67 |     using string    = basic_string<char>;
      |           ^~~~~~
In file included from include/boost/lexical_cast/detail/converter_lexical.hpp:52,
                 from include/boost/lexical_cast/try_lexical_convert.hpp:44,
                 from include/boost/lexical_cast.hpp:32,
                 from /nix/store/w9qnn4vw72fmiivd6p5hpiny504pz6yh-nix-2.8pre20220411_f7276bc-dev/include/nix/util.hh:14,
                 from ../libhydra/db.hh:5,
                 from hydra-evaluator.cc:1:
include/boost/container/container_fwd.hpp:280:31: note:   'boost::container::string'
In file included from hydra-evaluator.cc:2:
../libhydra/hydra-config.hh: In constructor 'HydraConfig::HydraConfig()':
../libhydra/hydra-config.hh:20:29: error: 'string' was not declared in this scope
   20 |                 line = trim(string(line, 0, line.find('#')));
      |                             ^~~~~~
../libhydra/hydra-config.hh:20:29: note: suggested alternatives:
In file included from /nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/string:39,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/internal/encodings.hxx:16,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array.hxx:18,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array:4,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/pqxx:2,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
/nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/bits/stringfwd.h:79:33: note:   'std::string'
   79 |   typedef basic_string<char>    string;
      |                                 ^~~~~~
In file included from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/internal/encodings.hxx:16,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array.hxx:18,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/array:4,
                 from /nix/store/vsv4ppgvqbx5n6liy23qvymb94xb3zs3-libpqxx-6.4.8/include/pqxx/pqxx:2,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
/nix/store/19xbyxc31snlk60cil7cx6l4xw126ids-gcc-11.2.0/include/c++/11.2.0/string:67:11: note:   'std::pmr::string'
   67 |     using string    = basic_string<char>;
      |           ^~~~~~
In file included from include/boost/lexical_cast/detail/converter_lexical.hpp:52,
                 from include/boost/lexical_cast/try_lexical_convert.hpp:44,
                 from include/boost/lexical_cast.hpp:32,
                 from /nix/store/w9qnn4vw72fmiivd6p5hpiny504pz6yh-nix-2.8pre20220411_f7276bc-dev/include/nix/util.hh:14,
                 from ../libhydra/db.hh:5,
                 from hydra-evaluator.cc:1:
include/boost/container/container_fwd.hpp:280:31: note:   'boost::container::string'
make[3]: *** [Makefile:442: hydra_evaluator-hydra-evaluator.o] Error 1
make[3]: Leaving directory '/build/source/src/hydra-evaluator'
make[2]: *** [Makefile:362: all-recursive] Error 1
make[2]: Leaving directory '/build/source/src'
make[1]: *** [Makefile:417: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:347: all] Error 2
```

**Comment by ivan** [comment]
I managed to build hydra-unstable on master in https://github.com/ivan/nixpkgs/commits/hydra-on-master

I applied most of #160202 (the later commits I could not apply), updated Hydra again, reverted nixUnstable twice, and added some missing build inputs.

**Comment by ivan** [comment]
Was this fixed by #172242?

**Comment by Mindavi** [comment]
Yes, if it wasn't I messed up :)
